### DISCUSSION TITLE
What is the XML profile configuration that corresponds to ROS_STATIC_PEERS=a.b.c.d

### DISCUSSION THREAD

**Discussion by razr** [eProsima/Fast-DDS#3811] [Q&A]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

Hello @MiguelCompany 

I want to communicate between the docker container and my host, both running ROS2 `humble` and Ubuntu 22.04.
It works fine on `rolling` with `ROS_STATIC_PEERS`, but does not work on `humble`.

 ```bash
# Docker side
docker run -ti -h ros2native -v $PWD:/work <image>
cd ros2_native/install/
source setup.bash

env | grep ROS
ROS_VERSION=2
ROS_PYTHON_VERSION=3
ROS_AUTOMATIC_DISCOVERY_RANGE=SUBNET
ROS2_WS=/opt/ros2_ws
ROS_DISTRO=rolling

export ROS_STATIC_PEERS=172.17.0.1
ros2 run demo_nodes_cpp listener
[INFO] [1692351185.411086291] [listener]: I heard: [Hello World: 1]
[INFO] [1692351186.410639363] [listener]: I heard: [Hello World: 2]

# Host side
cd ros2_native/install/
source setup.bash
export ROS_STATIC_PEERS=172.17.0.2
ros2 run demo_nodes_cpp talker
[INFO] [1692351185.410282463] [talker]: Publishing: 'Hello World: 1'
[INFO] [1692351186.410174763] [talker]: Publishing: 'Hello World: 2
```

What would be the proper FastDDS XML configuration for it on `humble`? I have looked [here](https://fast-dds.docs.eprosima.com/en/latest/fastdds/ros2/ros2_configure.html) but couldn't find it.

### Current behavior

communication between docker and host both running ROS 2 `humble` `talker/listener` does not work

### Steps to reproduce

build ROS 2 `humble` on Ubuntu 22.04 and run the `talker` example inside docker and the `listener` example on the host.

### Fast DDS version/commit

humble

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

Default configuration, UDPv4 & SHM

### Additional context

Ubuntu 22.04

**Comment by JLBuenoLopez** [comment]
Thanks for your contribution, @razr 

`ROS_STATIC_PEERS` was introduced in ROS 2 Iron so it is not available in ROS 2 Humble. You may consider upgrading to the latest ROS 2 version to leverage its latest improvements.

This ROS 2 environment variable is equivalent to the [initial peers Fast DDS feature](https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#initial-peers) (see the [RMW implementation code](https://github.com/ros2/rmw_fastrtps/blob/rolling/rmw_fastrtps_shared_cpp/src/participant.cpp#L243) for reference).

Also, I am moving this ticket to the Q&A discussion forum as per [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#issues-and-support)

**Comment by razr** [comment]
Thanks much @JLBuenoLopez-eProsima for your quick reply. Let me try it out

**Comment by razr** [comment]
@JLBuenoLopez-eProsima 
this is my `fastdds-docker.xml` file from https://fast-dds.docs.eprosima.com/en/latest/fastdds/discovery/simple.html#initial-peers

```xml
<!--
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
-->
    <participant profile_name="initial_peers_example_profile" is_default_profile="true">
        <rtps>
            <builtin>
                <initialPeersList>
                    <locator>
                        <udpv4>
                            <address>172.17.0.1</address>
                            <port>7412</port>
                        </udpv4>
                    </locator>
                </initialPeersList>
            </builtin>
        </rtps>
    </participant>
```

If I try to run my example with it, I get the following error

```bash
wruser@ros2native:/work/ros2_native/install$ env | grep FAST
FASTRTPS_DEFAULT_PROFILES_FILE=/work/fastdds-docker.xml

wruser@ros2native:/work/ros2_native/install$ ros2 run demo_nodes_cpp listener
2023-08-22 11:05:26.319 [XMLPARSER Error] Not found root tag -> Function parseXML
2023-08-22 11:05:26.319 [XMLPARSER Error] Error parsing '/work/fastdds-docker.xml' -> Function loadXMLFile

wruser@ros2native:/work/ros2_native/install$ env | grep ROS
ROS_VERSION=2
ROS_PYTHON_VERSION=3
ROS2_WS=/opt/ros2_ws
ROS_LOCALHOST_ONLY=0
ROS_DISTRO=humble
```

**Reply by JLBuenoLopez** [reply]
Please, remove the XML comment `<!-- -->` and remember to close the profiles tag too, `</profiles>` at the end of the file. Refer to [Fast DDS documentation](https://fast-dds.docs.eprosima.com/en/latest/fastdds/xml_configuration/xml_configuration.html) for more information about XML configuration files.

**Comment by razr** [comment]
@JLBuenoLopez-eProsima thank you for reply

I have fixed my XML file but still, there is no connection between the docker and host

## Docker

```bash
wruser@ros2native:/work/ros2_native/install$ hostname -I
172.17.0.4

wruser@ros2native:/work/ros2_native/install$ cat /work/fastdds-docker.xml
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <participant profile_name="initial_peers_example_profile" is_default_profile="true">
        <rtps>
            <builtin>
                <initialPeersList>
                    <locator>
                        <udpv4>
                            <address>172.17.0.1</address>
                            <port>7412</port>
                        </udpv4>
                    </locator>
                </initialPeersList>
            </builtin>
        </rtps>
    </participant>
</profiles>

wruser@ros2native:/work/ros2_native/install$ export FASTRTPS_DEFAULT_PROFILES_FILE=/work/fastdds-docker.xml

wruser@ros2native:/work/ros2_native/install$ ros2 run demo_nodes_cpp listener
```

## Host

```bash
$ ifconfig docker0
docker0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.17.0.1  netmask 255.255.0.0  broadcast 172.17.255.255

$ ping 172.17.0.4
PING 172.17.0.4 (172.17.0.4) 56(84) bytes of data.
64 bytes from 172.17.0.4: icmp_seq=1 ttl=64 time=0.038 ms
64 bytes from 172.17.0.4: icmp_seq=2 ttl=64 time=0.034 ms


$ export FASTRTPS_DEFAULT_PROFILES_FILE=$HOME/github/razr/vxworks7-ros2-build/fastdds.xml

$ cat $HOME/github/razr/vxworks7-ros2-build/fastdds.xml
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <participant profile_name="initial_peers_example_profile" is_default_profile="true">
        <rtps>
            <builtin>
                <initialPeersList>
                    <locator>
                        <udpv4>
                            <address>172.17.0.4</address>
                            <port>7412</port>
                        </udpv4>
                    </locator>
                </initialPeersList>
            </builtin>
        </rtps>
    </participant>
</profiles>

$ ros2 run demo_nodes_cpp talker
[INFO] [1692708772.697120937] [talker]: Publishing: 'Hello World: 1'
[INFO] [1692708773.697069323] [talker]: Publishing: 'Hello World: 2'
[INFO] [1692708774.697118381] [talker]: Publishing: 'Hello World: 3'
```

**Reply by JLBuenoLopez** [reply]
You may be interested in the following [Vulcanexus tutorial](https://docs.vulcanexus.org/en/latest/rst/tutorials/core/qos/initial_peers/initial_peers.html?highlight=initial%20peers).
